crates/resolver/src/config.rs

```diff
@@ -906,6 +906,26 @@ impl Default for ServerOrderingStrategy {
     }
 }
 
+/// Whether the system hosts file should be respected by the resolver.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[cfg_attr(feature = "serde-config", derive(Serialize, Deserialize))]
+pub enum ResolveHosts {
+    /// Always attempt to look up IP addresses from the system hosts file.
+    /// If the hostname cannot be found, query the DNS.
+    Always,
+    /// The DNS will always be queried.
+    Never,
+    /// Use local resolver configurations only when this resolver is not used in
+    /// a DNS forwarder. This is the default.
+    Auto,
+}
+
+impl Default for ResolveHosts {
+    fn default() -> Self {
+        Self::Auto
+    }
+}
```

Comment:
With the current MSRV, I believe this can derived.

Reply:
Good point, done.